status/pricing.py

```diff
@@ -409,80 +97,49 @@ class PricingExchangeRatesDataHandler(PricingBaseHandler):
-        exch_rates['EUR_in_SEK'] = '{:.2f}'.format(float(exch_rates['EUR_in_SEK']))
-
-        t = self.application.loader.load('pricing_quote.html')
-        self.write(t.generate(gs_globals=self.application.gs_globals,
-                              user=self.get_current_user(),
-                              exch_rates=exch_rates))
+    def _fetch_exchange_rates(self, date=None):
+        """Internal method to fetch exchange rates from StatusDB
 
+        :param date: string in format 'YYYY-MM-DD'
+        :return: dictionary with keys 'USD_in_SEK', 'EUR_in_SEK' and
+        'Issued at'.
 
-class PricingQuoteTbodyHandler(PricingBaseHandler):
-    """ Serves a tbody specificly for /pricing_quote to be generated dynamically.
+        Returns the most recent rates prior to date given by parameter `date`
+        If parameter `date` is not supplied, the current timestamp is used.
+        """
 
-    Loaded through e.g.:
-        /pricing_quote_tbody?date=2019-03-23
+        if date is not None:
+            dt = self._validate_date_string(date)
+        else:
+            dt = datetime.datetime.now()
 
-    The parameter discontinued can be used to also show products where
-        the status is not 'Available'
+        str_format_date = dt.strftime('%Y-%m-%d')
+        view = self.application.pricing_exchange_rates_db.view(
+                'entire_document/by_date',
+                startkey=str_format_date,
+                descending=True,
+                limit=1
+            )
 
-    """
+        result = {}
+        result['USD_in_SEK'] = view.rows[0].value['USD_in_SEK']
+        result['EUR_in_SEK'] = view.rows[0].value['EUR_in_SEK']
+        result['Issued at'] = view.rows[0].value['Issued at']
+        return result
 
     def get(self):
-        version = self.get_argument('version', None)
         date = self.get_argument('date', None)
-        discontinued = self.get_argument('discontinued', None)
-
-        products = self.get_product_prices(None, version=version,
-                                           date=date,
-                                           discontinued=discontinued,
-                                           pretty_strings=True)
-
-        products = [product for id, product in products.items()]
-
-        components = self.get_component_prices(component_id=None,
-                                               version=version,
-                                               date=date,
-                                               pretty_strings=True)
-
-        t = self.application.loader.load('pricing_quote_tbody.html')
-        self.write(t.generate(gs_globals=self.application.gs_globals,
-                              user=self.get_current_user(),
-                              products=products,
-                              components=components,
-                              version=version))
 
+        if date is not None:
```

Comment:
Do we need this check? `date `is set to `None` above and `_fetch_exchange_rates` handles that case anyway

Reply:
You're absolutely right. Fixed!